orderer/common/multichannel/registrar.go

```diff
@@ -719,11 +721,11 @@ func (r *Registrar) JoinChannel(channelID string, configBlock *cb.Block, isAppCh
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
-	if removalStatus, ok := r.pendingRemoval[channelID]; ok {
-		if removalStatus {
-			return types.ChannelInfo{}, types.ErrChannelPendingRemoval
-		} else {
+	if status, ok := r.pendingRemoval[channelID]; ok {
+		if status.Status == types.StatusFailure {
 			return types.ChannelInfo{}, types.ErrChannelRemovalFailure
+		} else {
+			return types.ChannelInfo{}, types.ErrChannelPendingRemoval
```

Comment:
Style nit: no need for the else here.
```suggestion
		if status.Status == types.StatusFailure {
			return types.ChannelInfo{}, types.ErrChannelRemovalFailure
		} 
	       return types.ChannelInfo{}, types.ErrChannelPendingRemoval
```

Reply:
🧹